docs/deployment.md

```diff
@@ -95,6 +95,29 @@ Firestore event          | DocumentEventData     | --trigger-event providers/clo
 >   available in the `FirestoreEvent` class via the `Wildcards` property. This is subject to change,
 >   as it's inconsitent with other Functions Frameworks.
 
+### Triggering an HTTP function with a Cloud Pub/Sub push subscription
+
+HTTP functions can work as the endpoints for [Cloud Pub/Sub push
+subscriptions](https://cloud.google.com/pubsub/docs/push). If your
+function implements `ICloudEventFunction` or
+`ICloudEventFunction<MessagePublishedData>`, the Functions Framework
+will adapt the incoming HTTP request to present it to your function
+as a CloudEvent, as if you had deployed via `--trigger-topic`. The
+requests for push subscriptions do not contain topic data, but if
+you create the push subcription with a URL of
+`https://<your-function>/projects/<project-id>/topics/<topic-id>`
+then the Functions Framework will infer the topic name from the path
+of the HTTP request. If the topic name cannot be inferred
+automatically, a topic name of
+`projects/unknown-project!/topics/unknown-topic!` will be used in
```

Comment:
It seems like we aren't consistent about what we use here across languages.

- [In Node.js](https://github.com/GoogleCloudPlatform/functions-framework-nodejs/blob/master/test/integration/cloud_event.ts#L188) we just drop it and make the CE subject `//pubsub.googleapis.com/`
- [In Ruby](https://github.com/GoogleCloudPlatform/functions-framework-ruby/blob/main/test/test_legacy_event_converter.rb#L111) we set it to `//pubsub.googleapis.com/UNKNOWN_PUBSUB_TOPIC`

Also, I **think** it is might be possible to always extract the project name from the subscription in the payload.

Reply:
> It seems like we aren't consistent about what we use here across languages.

Mmm. The reason I chose this is to keep structure that will work in the absence of "real" data, but make it obvious that it's *not* real data. That may or may not be a good choice. Let's discuss this more before actually releasing.

> Also, I think it is might be possible to always extract the project name from the subscription in the payload.

That assumes that the subscription is in the same project as the topic, which may not be the case. (I *think* cross-project PubSub subscriptions are a thing, with appropriate IAM. I could be wrong.) For the emulator that doesn't matter, but for the genuine PubSub push notifications, I think it's better to give either "obviously wrong" data or "user-specified data" (the URL path) rather than "looks plausible but may be wrong" data.

We should definitely talk about that part more. I'm going to merge this as-is, but file a new issue so that we don't forget about it before the next release.